### PR TITLE
fix(rust, python): fix rolling weighted mean

### PR DESCRIPTION
**Pull request by magarick**
As discussed in https://github.com/pola-rs/polars/issues/8941 the rolling weighted mean and variance aren't computed correctly. This fixes it for the mean.

### CODE REVIEW

File: polars/polars-arrow/src/kernels/rolling/no_nulls/mean.rs [hunk 1]
```diff
@@ -35,41 +35,35 @@ pub fn rolling_mean<T>(
...
-                &weights,
-            )
-        }
-        (false, Some(weights)) => {
-            let weights = no_nulls::coerce_weights(weights);
+        Some(weights) => {
+            // A weighted mean is a weighted sum with normalized weights
+            let mut wts = no_nulls::coerce_weights(weights);
+            let wsum = wts.iter().fold(T::zero(), |acc, x| acc + *x);
+            assert_ne!(
```
Comment (ritchie46) [inline polars/polars-arrow/src/kernels/rolling/no_nulls/mean.rs:54]: We should return an error, or maybe do the weights validation a level higher in the function that does returns a `PolarsResult`.

Comment (magarick) [reply polars/polars-arrow/src/kernels/rolling/no_nulls/mean.rs:54]: This function gets called from several higher level functions so I'd have to replicate the check and I'm not nuts about that.
Would returning an error require changing the function signature here? If so we'd have to change all the rolling kernel signatures I think.

Comment (ritchie46) [reply polars/polars-arrow/src/kernels/rolling/no_nulls/mean.rs:54]: > Would returning an error require changing the function signature here? If so we'd have to change all the rolling kernel signatures I think.

That's also fine. We can wrap the results in an `Ok`.

**Comment by magarick** [comment]
Alright. I wrapped everything in a `PolarsResult` since checking right before computing seemed like the clearest way to me.
Two caveats:
1. Because quantiles are special-cased they don't get the `PolarsResult` treatment. But it does raise the fact that it's possible to avoid that now and get clearer code since the rolling functions take `DynArgs`.
2. Weighted rolling variance is still wrong. You can decide which weight interpretation you want to go with there (see the original issue if you feel like it). I think normalized weights like we're doing for the mean is fine though. "Unbiasedness" doesn't make sense in this rolling context anyway.

**Comment by ritchie46** [comment]
> Because quantiles are special-cased they don't get the PolarsResult treatment. But it does raise the fact that it's possible to avoid that now and get clearer code since the rolling functions take DynArgs.

Yes, they could use the same architecture now. 

Thanks for the fix @magarick!

The failing tests are unrelated, so merging in.

**Pull request merged**